sktime/networks/base.py

```diff
@@ -25,3 +28,154 @@ def build_network(self, input_shape, **kwargs):
         output_layer : a keras layer
         """
         ...
+
+
+class BaseDeepNetworkPyTorch(BaseForecaster, ABC):
+    """Abstract base class for deep learning networks using torch.nn."""
+
+    _tags = {
+        "python_dependencies": "torch",
+    }
+
+    def __init__(self):
+        super().__init__()
+
+    def _fit(self, y, X, fh):
+        """Fit the network.
+
+        Changes to state:
+            writes to self._network.state_dict
+
+        Parameters
+        ----------
+        X : iterable-style or map-style dataset
+            see (https://pytorch.org/docs/stable/data.html) for more information
+        """
+        dataloader = self.build_pytorch_train_dataloader(y)
+        self.network.train()
+
+        for _ in range(self.num_epochs):
+            for x, y in dataloader:
+                y_pred = self.network(x)
+                loss = self._criterion(y_pred, y)
+                self._optimizer.zero_grad()
+                loss.backward()
+                self._optimizer.step()
+
+        # self._fh = self.pred_len
+
+    def _predict(self, X, **kwargs):
+        """Predict with fitted model."""
+        from torch import cat
```

Comment:
If an fh is passed here that does not fit the fh provided in `fit`, we should probably raise an exception or at least a warning, since they have to match. @fkiraly what would be the best way to do it in sktime? 

Furthermore, future work could relax this a bit I adding the possibility of calling the network in an autoregressive way.

Reply:
forecasters methods (public and private) should not have kwargs, although this is a network, so the syntax is completely open. Having said that, a `fh` in a network doesn't feel right, design-wise.